docs/proposals/metadata.md

```diff
@@ -0,0 +1,172 @@
+etc. Automatic generation of a PodTemplate is also required for
+DeploymentConfigs.
+
+## Constraints and Assumptions
+
+This document only defines the metadata needed by the current set of use cases.
+Additional metadata and/or use cases may be added in the future.
+
+The metadata should be set using the [LABEL](https://docs.docker.com/reference/builder/#label)
+Docker instruction.
+
+Once the Docker image is built the metadata can be obtained by running the
+`docker inspect` command. The metadata should be available in the `Labels`
+section.
+
+For more default about conventions or guidelines about LABEL, visit:
+https://docs.docker.com/userguide/labels-custom-metadata
+
+> The *LABEL* instruction is supported in Docker starting from the version
+> 1.6.0. This instruction wont work on older Docker.
+
+## Use cases
+
+1. As an author of Docker image that is going to be consumed by the OpenShift
+   platform, I want to express what categories/tags my Docker image will belongs
+   to, so the platform then can then consume the tags to improve the generation
+   workflow based on them.
+
+2. As an user of OpenShift I want to get reliable suggestions about services
+   that the Docker image I'm going to use might require to operate properly. As
+   an author of the Docker image, I need to have a way to record what services my
+   Docker image want to consume.
+
+3. As an author of the Docker image, I need to have a way to indicate whether
+   the container started from my Docker image does not support scaling.
+   The UI should then communicate this information with the end consumers.
+
+4. As an author of the Docker image, I need to have a way to indicate what
+   additional service(s) my Docker image might need to work properly, so the UI
+   or generation tools can suggest them to the end users.
+
+## Namespaces
+
+The LABEL names should typically be namespaced. The namespace should be set
+accordingly to reflect the project that is going to pick up the labels and use
+them. For OpenShift the namespace should be set to `openshift.io/` and for
+Kubernetes the namespace is `k8s.io/`. For simple labels, like `displayName` or
+`description` there might be no namespace set if they end up as standard in
+Docker.
+
+## Image Metadata
+
+Name                                  | Type     | Target Namespace |
+--------------------------------------|--------- | ------------------
+[`tags`](#tags)                       | []string | openshift.io
+[`wants`](#wants)                     | []string | openshift.io
+[`display-name`](#display-name)       |   string | k8s.io
+[`description`](#description)         |   string | k8s.io
+[`expose-services`](#expose-services) | []string | openshift.io
+[`non-scale`](#non-scale)             |     bool | openshift.io
+[`min-cpu`](#min-cpu)                 |   string | openshift.io(?)
+[`min-memory`](#min-memory)           |   string | openshift.io(?)
```

Comment:
what happens when a docker 1.5 client pulls an image with labels?  

also we can't remove the existing ENV "labels" until we update the openshift code to look at the docker labels instead of the ENV variables, so this probably needs to be a 3 step process.  (and we can't do it until we've moved to a 1.6 prereq)